python/pyspark/pandas/indexes/base.py

```diff
@@ -926,7 +928,15 @@ def struct_to_array(scol: Column) -> Column:
                 field_names = result._internal.spark_type_for(
                     scol
                 ).fieldNames()  # type: ignore[attr-defined]
-                return F.array([scol[field] for field in field_names])
+                field_types = [
+                    result._internal.spark_type_for(scol[field]) for field in field_names
+                ]
+
+                has_str = any(isinstance(t, StringType) for t in field_types)
+                if is_ansi_mode_enabled and has_str:
```

Comment:
Need to pass the Spark session to `is_ansi_mode_enabled`; otherwise it's always `True`?

Reply:
Good catch, thanks!